fairlearn/utils/_input_validation.py

```diff
@@ -84,18 +85,20 @@ def _validate_and_reformat_input(
     elif expect_y:
         raise ValueError(_MESSAGE_Y_NONE)
     else:
-        X = check_array(X)
+        X = check_array(X, dtype=None, force_all_finite=False)
 
     sensitive_features = kwargs.get(_KW_SENSITIVE_FEATURES)
-    if sensitive_features is None:
-        raise ValueError(_MESSAGE_SENSITIVE_FEATURES_NONE)
+    if sensitive_features is not None:
```

Comment:
The previous code might assume that `sensitive_features` are always provided, so please introduce `expect_sensitive_features` which would default to `True`.